src/xbot_monitoring.cpp

```diff
@@ -57,6 +59,27 @@ std::string external_mqtt_topic_prefix = "";
 std::string external_mqtt_port = "";
 std::string version_string = "";
 
+// separator for action_id, request_id and payload/response of actions
+const char ACTION_SEP = '|';
```

Comment:
why not use json on the wire instead of custom format?

Reply:
I considered that, and if you want, I can change it to JSON. My thoughts were:
- Don't want to break existing apps, which is achieved by making the request_id and payload optional. Therefore a simple `mower_logic/reset_emergency` still works. Requiring the action to be a JSON like `{"action": "mower_logic/reset_emergency", "request_id": "123-456", "payload": [1, 2, 3]}` would break apps. We could check whether the first character is a `{` though, and if not convert it to `{"action": "<message>"}`, so that's no blocker.
- We can't pass the parsed JSON object via ROS messages. That means we have two options: 
  - Pass the MQTT payload string unmodified to the callers. Publishing type would be string as before, but the downside is that every client has to decode the JSON on their own, including error handling. Also, I like in my approach that there are three fields with a clear purpose, rather than the generic string field. That would make it easier (from my point of view) to request an action within ROS itself.
  - Decode the JSON in xbot_monitoring and fill the three fields in the ROS message with it. Problem: What if the payload is a JSON structure itself, as in the example above? Then xbot_monitoring would have to dump that sub-structure into a string again which gets filled into the payload field in the ROS message. Or the client would have to pass their payload as string only. Both are quite ugly.

On the other hand, I don't expect any ambiguities with my proposal, as neither action_id nor request_id will have the `|` character. It's easy to concatenate and split these strings. And JSON decoding only needs to be done if really required.